src/helpers/__tests__/snippet-test.ts

```diff
@@ -122,4 +123,33 @@ describe('snippet', () => {
       `"Nested <mark class=\\"ais-Snippet-highlighted\\">Amazon</mark> name"`
     );
   });
+
+  test('with nested attribute as array', () => {
+    expect(
+      snippet({
+        attribute: ['meta', 'name'],
+        hit,
+      })
+    ).toMatchInlineSnapshot(
+      `"Nested <mark class=\\"ais-Snippet-highlighted\\">Amazon</mark> name"`
+    );
+  });
+
+  test('with array attribute', () => {
+    expect(
+      snippet({
+        attribute: 'categories.1',
+        hit,
+      })
+    ).toMatchInlineSnapshot(`"Streaming Media Players"`);
+  });
+
+  test('with array attribute as array', () => {
+    expect(
+      snippet({
+        attribute: ['categories', 1],
```

Comment:
Since it's a path it makes sense to _only_ accept strings:

```
attribute: ['categories', '1'],
```

It'll make the type more simple.

Reply:
either version works, since it gets coerced anyway when you do a key check, but I'm ok reverting my previous commit